### DISCUSSION TITLE
Dagster-celery worker does not know the run id

### DISCUSSION THREAD

**Discussion by mmarionncern** [dagster-io/dagster#10883] [Q&A] [answered]
Dear dagster experts,

I am trying to setup a small VM dagster/celery farm on two machines, one with my job scheduler, and one with my celery worker.
The following code is present on both machines :

```
from dagster_celery import celery_executor
from dagster import repository, job, op, Field, Noneable, StringSource

from dagster_celery.tasks import create_task

ce = celery_executor.configured({"broker":'redis://*******',
                                 "backend":'redis://*******'})

@op(tags={'dagster-celery/queue': 'queue_A'})
def not_much():
    return

@job(executor_def=ce)
def parallel_job():
   for i in range(1):
       not_much.alias("not_much_" + str(i))()

@repository
def test_repo():
    return [parallel_job]
```

The two machines communicate well and the scheduler send the proper configuration to the worker :
`[2022-12-05 15:10:27,818: WARNING/ForkPoolWorker-2] ExecuteStepArgs(pipeline_origin=PipelinePythonOrigin(pipeline_name='parallel_job', repository_origin=RepositoryPythonOrigin(executable_path='/usr/bin/python3', code_pointer=FileCodePointer(python_file='/home/ubuntu/celery/repository.py', fn_name='test_repo', working_directory='/home/ubuntu/celery'), container_image=None, entry_point=['dagster'], container_context=None)), pipeline_run_id='9550a351-065b-49db-819c-8344b1f6996d', step_keys_to_execute=['not_much_0'], instance_ref=InstanceRef(local_artifact_storage_data=ConfigurableClassData(module_name='dagster._core.storage.root', class_name='LocalArtifactStorage', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa\n'), compute_logs_data=ConfigurableClassData(module_name='dagster._core.storage.local_compute_log_manager', class_name='LocalComputeLogManager', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa/storage\n'), scheduler_data=ConfigurableClassData(module_name='dagster._core.scheduler', class_name='DagsterDaemonScheduler', config_yaml='{}\n'), run_coordinator_data=ConfigurableClassData(module_name='dagster._core.run_coordinator', class_name='DefaultRunCoordinator', config_yaml='{}\n'), run_launcher_data=ConfigurableClassData(module_name='dagster', class_name='DefaultRunLauncher', config_yaml='{}\n'), settings={}, run_storage_data=ConfigurableClassData(module_name='dagster._core.storage.runs', class_name='SqliteRunStorage', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa/history/\n'), event_storage_data=ConfigurableClassData(module_name='dagster._core.storage.event_log', class_name='SqliteEventLogStorage', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa/history/runs/\n'), schedule_storage_data=ConfigurableClassData(module_name='dagster._core.storage.schedules', class_name='SqliteScheduleStorage', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa/schedules\n'), custom_instance_class_data=None, storage_data=ConfigurableClassData(module_name='dagster._core.storage.sqlite_storage', class_name='DagsterSqliteStorage', config_yaml='base_dir: /home/ubuntu/celery/tmpo0k0ujxa\n'), secrets_loader_data=None), retry_mode=<RetryMode.DEFERRED: 'deferred'>, known_state=KnownExecutionState(previous_retry_attempts={}, dynamic_mappings={}, step_output_versions=[], ready_outputs=set(), parent_state=None), should_verify_step=False)`

but the execution fails as the worker does not know the run id :
`[2022-12-05 15:10:28,046: ERROR/ForkPoolWorker-2] Task execute_plan[97b89c20-e9be-4976-9411-d095aea366a1] raised unexpected: CheckError('Invariant failed. Description: Could not load run 9550a351-065b-49db-819c-8344b1f6996d'`

Looking at the documentation and the git hub discussion I do not see obvious answer to that problem. Am I missing something obvious or important?

Many thanks by advance

**Comment by alangenfeld** [comment, marked as answer]
https://docs.dagster.io/deployment/dagster-instance

you will have to ensure that the two machines can see the same Dagster storage - likely by switching to a database like postgres.

**Comment by mmarionncern** [comment]
Thanks a lot, this was indeed the missing piece
